tests/main/core-persistent-journal/task.yaml

```diff
@@ -0,0 +1,33 @@
+summary: Test journal.persistent core config option
+
+systems: [ubuntu-core-*]
+
+execute: |
+    echo "Wait for first boot to be done"
+    while ! snap changes | grep -q "Done.*Initialize system state"; do sleep 1; done
+
+    echo "Sanity check, persistent journal is not available by default"
+    not test -e /var/log/journal
+
+    echo "Check that persistent journal can be enabled"
+    snap set core journal.persistent=true
+    test -e /var/log/journal/
+    test -e /var/log/journal/.snapd-created
+
+    # this check relies on *anything* getting logged; enabling persistent
+    # journal writes an entry about journal size, which should be sufficient.
+    MACHINE_ID=$(cat /etc/machine-id)
+    retry-tool -n30 --wait 1 test -e "/var/log/journal/$MACHINE_ID"
+
+    echo "Check that persistent journal can be disabled"
+    snap set core journal.persistent=false
+    not test -e /var/log/journal
+
+    echo "Check that journal log is not removed if managed manually"
+    mkdir /var/log/journal
+    snap set core journal.persistent=true
+    not test -e /var/log/journal/.snapd-created
+    snap set core journal.persistent=false 2>&1 | MATCH "the /var/log/journal directory was not created by snapd"
+    # the journal dir was not removed
+    test -e /var/log/journal/
```

Comment:
does it mean the test finishes with persistent logging on?  should we turn it off in a restore ?

Reply:
Good point, added to restore.